meta-balena-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -222,18 +222,6 @@ if [ -d /sys/firmware/efi ]; then
                     /usr/bin/efi-updatevar -f "${EXTERNAL_DEVICE_BOOT_PART_MOUNTPOINT}/balena-keys/${e}.auth" "${e}"
                 fi
             done
-            # Check that we're in user mode
-            SETUPMODE_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SetupMode \
-                | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
-            if [ "${SETUPMODE_VAR}" -eq "1" ]; then
-                # Setting up keys hasn't disabled setup mode, try a reboot into flasher
-                info "Rebooting flasher after secure mode setup to boot in secure boot mode."
-                # Make sure to reboot into the installer
-                bootcurrent=$(efibootmgr | grep BootCurrent | awk '{print $2}')
-                efibootmgr -n "${bootcurrent}"
-                sync
-                reboot -f
-            fi
```

Comment:
I think we should keep the reboot in to ensure we are actually able to enable secure boot. Having the check in the initrd is only making it more annoying to break in, not more secure. If we provision an encrypted system with secure boot not enforced in HW, you can just go to the initrd, remove the check and it will work.

Reply:
@mtoman With the boot entry change fixing the same issue, should we close this PR?